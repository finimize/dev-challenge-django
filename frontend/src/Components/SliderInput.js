import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './SliderInput.css'

export default class SliderInput extends Component {

	constructor(props) {
		super(props)

		this.state = {
			value: props.defaultValue
		}
	}

	handleChange(e) {
		const value = e.target.value
		this.setState({value})
		this.props.onChange(e)
	}

	render() {
		const { name } = this.props
		const { value } = this.state

		return (
			<div className="fmz-slider">
				<p>{value}%</p>
				<input type="range"
					name={name}
					value={value}
					min={0}
					max={10}
					step={0.25}
					onChange={this.handleChange.bind(this)}/>
			</div>
		)
	}
}

SliderInput.propTypes = {
	defaultValue: PropTypes.number
}
