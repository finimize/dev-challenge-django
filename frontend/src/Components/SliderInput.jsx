import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './SliderInput.css';

export default class SliderInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: props.defaultValue,
    };
  }

  handleChange(e) {
    const { value } = e.target;
    const { onChange } = this.props;
    this.setState({ value });
    onChange(parseFloat(value));
  }

  render() {
    const { value } = this.state;

    return (
      <div className="fmz-slider">
        <p>
          {value}
%
        </p>
        <input
          type="range"
          value={value}
          min={0}
          max={10}
          step={0.25}
          onChange={(e) => { this.handleChange(e); }}
        />
      </div>
    );
  }
}

SliderInput.propTypes = {
  defaultValue: PropTypes.number,
  onChange: PropTypes.func,
};

SliderInput.defaultProps = {
  defaultValue: 0,
  onChange: () => {},
};
