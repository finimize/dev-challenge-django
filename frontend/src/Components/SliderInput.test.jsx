import React from 'react';
import TestRenderer from 'react-test-renderer';

import SliderInput from './SliderInput';

describe('SliderInput', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(
      <SliderInput value={5} onChange={() => {}} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
