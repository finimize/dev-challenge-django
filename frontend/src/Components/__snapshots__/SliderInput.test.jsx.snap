// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderInput renders correctly 1`] = `
<div
  className="fmz-slider"
>
  <p>
    0
    %
  </p>
  <input
    max={10}
    min={0}
    onChange={[Function]}
    step={0.25}
    type="range"
    value={0}
  />
</div>
`;
